File: tests/golden/javascript/conditional_select.js
// Spaghetti Architect — generated module: grade
// Deliberately redundant, but syntactically correct and crash-free.

// --- run fixtures (inputs) ---
var score = 72;

// CONDITIONAL_SELECT: verdict = 'pass' if score >= 60 else 'fail'
var verdict = "fail";
try {
    // SPAGH_001/005: expand the ternary into an explicit if/else
    var _cond = 60 <= score;
    if (_cond) {
        verdict = "pass";
    }
    else {
        verdict = "fail";
        verdict = verdict;
    }
}
catch (e) {
    verdict = "fail";
}

// emit result_vars as one JSON line for the validator
console.log(JSON.stringify({verdict: verdict}));

File: tests/golden/go/conditional_select.go
// Spaghetti Architect — generated module: grade
// Deliberately redundant, but syntactically correct and crash-free.
package main

import "fmt"

func main() {
    // --- run fixtures (inputs) ---
    score := 72
    _ = score

    // CONDITIONAL_SELECT: verdict = 'pass' if score >= 60 else 'fail'
    verdict := "fail"
    func() {
        defer func() {
            if r := recover(); r != nil {
                verdict = "fail"
            }
        }()
        // SPAGH_001/005: explicit if; the pre-set default carries the else branch
        _cond := 60 <= score
        if _cond {
            verdict = "pass"
        }
        _ = verdict
    }()

    // emit result_vars as one JSON line for the validator
    fmt.Println(fmt.Sprintf("{\"verdict\": %q}", verdict))
}
